lib/stsci/tools/bitmask.py

```diff
@@ -8,18 +9,399 @@
+    is obtained by summing input flags.
+
+    .. note::
+        In order to flip the bits of the returned bitmask,
+        for input of `str` type, prepend '~' to the input string. '~' must
+        be prepended to the *entire string* and not to each bit flag! For
+        input that is already a bitmask or a Python list of bit flags, set
+        `flip_bits` for `True` in order to flip the bits of the returned
+        bitmask.
+
+    Parameters
+    ----------
+    bit_flags : int, str, list, None
+        An integer bitmask or flag, `None`, a string of comma- or
+        '+'-separated list of integer bit flags, or a Python list of integer
+        bit flags. If `bit_flags` is a `str` and if it is prepended with '~',
+        then the output bitmask will have its bits flipped (compared to simple
+        sum of input flags). For input `bit_flags` that is already a bitmask
+        or a Python list of bit flags, bit-flipping can be controlled through
+        `flip_bits` parameter.
+
+    flip_bits : bool, None
+        Indicates whether or not to flip the bits of the returned bitmask
+        obtained from input bit flags. This parameter must be set to `None`
+        when input `bit_flags` is either `None` or a Python list of flags.
+
+    Returns
+    -------
+    bitmask : int or None
+        Returns and integer bit mask formed from the input bit value
+        or `None` if input `bit_flags` parameter is `None` or an empty string.
+        If input string value was prepended with '~' (or `flip_bits` was
+        set to `True`), then returned value will have its bits flipped
+        (inverse mask).
+
+    Examples
+    --------
+        >>> from stsci.tools.bitmask import interpret_bit_flags
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags(28))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('4,8,16'))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~4,8,16'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~(4+8+16)'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16]))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16], flip_bits=True))
+        '1111111111100011'
+
+    """
+    has_flip_bits = flip_bits is not None
+    flip_bits = bool(flip_bits)
+    allow_non_flags = False
+
+    if isinstance(bit_flags, int):
+        return (~bit_flags if flip_bits else bit_flags)
+
+    elif bit_flags is None:
+        if has_flip_bits:
+            raise TypeError(
+                "Keyword argument 'flip_bits' must be set to 'None' when "
+                "input 'bit_flags' is None."
+            )
+        return None
+
+    elif hasattr(bit_flags, '__iter__'):
+        pass
+
+    elif isinstance(bit_flags, six.string_types):
+        if has_flip_bits:
```

Comment:
@MSeifert04 I specifically do not want to allow "double-flip" as, I think, this will be a major source of ambiguity and confusion. I was also considering making ``flip_bits`` parameter a ``kwargs`` and forbid it for string input. However, I was not sure that having a parameter that is sometimes allowed and sometimes not allowed would be a better solution.